### ISSUE TITLE
Add a helper to allow Tokio to send signals to Child processes

### ISSUE THREAD

**Issue by strottos** [tokio-rs/tokio#3379]
If I spawn a child process using Tokio, e.g.
```
let process = Command::new("my_prog")
        .spawn()
        .unwrap();
```
then it would be helpful to be able to send a signal to that child process of my choosing. Something along the lines of:
```
process.send_signal(SignalKind::SIGINT);
```

My understanding is that signal sending is non-blocking (or at least a very fast syscall) so that this can be done using something from the `nix` crate say (which I think the process flag in Tokio brings in already, but I could be wrong).

You may need to consider what to do on Windows with this as well. Maybe these calls can just be ignored there, I'm not sure what's best offhand.

**Comment by olix0r** [comment]
If it's helpful, I had to [implement something like this](https://github.com/olix0r/linkerd-await/blob/57590fc9c808216a879f56be2c181d5353b397cc/src/main.rs#L124-L143) recently.

**Comment by ipetkov** [comment]
tokio does not depend on `nix` at all today, but fwiw it can be done with a `libc::kill` call as well  (though you should take care to not cache the child PID and only signal if `child.id().is_some()`, else it is possible to signal some *other* process if the pid got recycled).

If such an API was to be added it would probably need to be OS specific since the signal semantics (outside of killing the process outright) break down with Windows

**Comment by carlosb1** [comment]
I could try to implement some solution.. It can be [this](https://microsoft.github.io/windows-docs-rs/doc/bindings/windows/)  useful

**Comment by carlosb1** [comment]
I did a brief research about how we could implement this for Windows.. I don't develop in Windows so they are assumptions:
- How it was commented before, POSIX and Winstraightdows don't manage processes in the same way. For this reason, It doesn't seem straight to apply signals for windows
- Libc doesn't have support for some operations. In fact, the operation  `kill` doesn't seem available  for windows
- To manage Windows processes it seems necessary to use its winrt (library) [https://www.codeproject.com/Articles/18962/Kill-Application-Using-Win32-APIs] but It seems tricky and it should be done with some windows api like (this)[https://docs.rs/windows-win/2.4.0/windows_win/struct.Process.html#method.open] or like (this)[microsoft.github.io/windows-docs-rs/]

In short, for POSIX signals, It should be a problem but for Windows... I seems tricky
